### PR TITLE
feat: add sparkfun-promini-v5

### LINKED ISSUE #24: missing SparkFun ProMini 5V board template
Among the options for boards presented during the cargo generate step the SparkFun ProMini 5V is seemingly missing.

### PR DESCRIPTION
**Pull request by emilbonnek**
Following [this pr](https://github.com/Rahix/avr-hal/pull/435) I found there is no support for this in the cargo templating.

I have tried adding it here, but I am not totally sure its done correctly. 

(closes #24)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -38,6 +38,8 @@ rev = "7dfa6d322b9df98b2d98afe0e14a97afe0187ac1"
     features = ["arduino-uno"]
   {%- when "SparkFun ProMicro" -%}
     features = ["sparkfun-promicro"]
+  {%- when "SparkFun ProMini 5v" -%}
+    features = ["sparkfun-promicro-5v"]
```
Comment (Rahix) [inline Cargo.toml]: This needs to be fixed:

```suggestion
    features = ["sparkfun-promini-5v"]
```

File: src/main.rs [hunk 2]
```diff
@@ -32,7 +32,10 @@ fn main() -> ! {
         "Adafruit Trinket Pro"
       -%}
     let mut led = pins.d13.into_output();
-      {%- when "SparkFun ProMicro" -%}
+      {%- when
+        "SparkFun ProMicro",
+        "SparFun ProMini 5v"
```
Comment (Rahix) [inline src/main.rs]: Also needs fixing:

```suggestion
        "SparkFun ProMini 5v"
```

File: src/main.rs [hunk 3]
```diff
@@ -32,7 +32,10 @@ fn main() -> ! {
...
+      -%}
     let mut led = pins.led_rx.into_output();
     {%- endcase %}
```
Comment (Rahix) [inline src/main.rs]: You've added the _ProMini_ to the wrong block here.  It needs the `d13` pin for the LED which means you should add the ProMini to the block above where `d13` is used.

**Review by Rahix — CHANGES_REQUESTED** [review]
Hi,

thanks a lot for the contribution!

I gave it a test using

```terminal
cargo generate --git https://github.com/emilbonnek/avr-hal-template -b patch-1
```

but there are still some issues.  First of all, the template was still referencing an old version of `avr-hal` which I already fixed in commit 6bac4dd3c7a35098c18da19baf7695283f3d9549.  You'll need to update your branch to include this change (rebase on latest `main`).

Beyond that, please see my comments below.

**Comment by emilbonnek** [comment]
Sorry about those!

@Rahix Let me know if that did it.

**Comment by Rahix** [comment]
Nothing to be sorry about, code review cycles are a normal part of any open-source contribution ;)

Just for the future - you could have tested this yourself by running

```bash
cargo generate --git https://github.com/emilbonnek/avr-hal-template -b patch-1 --name sparkfun-promini-v5-test
cd sparkfun-promini-v5-test/
cargo build
```

**Review by Rahix — APPROVED** [review]
Looks good to me, at least I can compile the example program now.

**Comment by emilbonnek** [comment]
> Nothing to be sorry about, code review cycles are a normal part of any open-source contribution ;)

Oh yeah, I know. I meant the spelling mistakes. I was on the phone when writing and I could have put in a little more effort so you wouldn't have to check these.

> Looks good to me, at least I can compile the example program now.

👍Yay. Cheers! and thanks for making this project.

**Pull request merged**